src/Microsoft.Health.Fhir.Core/Features/Operations/BulkUpdate/Handlers/CreateBulkUpdateHandler.cs

```diff
@@ -84,13 +84,19 @@ public async Task<CreateBulkUpdateResponse> Handle(CreateBulkUpdateRequest reque
             var dateCurrent = new PartialDateTime(Clock.UtcNow);
             searchParameters.Add(Tuple.Create("_lastUpdated", $"lt{dateCurrent}"));
 
+            // remnove maxCount from searchParameters
+            searchParameters.RemoveAll(t => t.Item1.Equals(KnownQueryParameterNames.MaxCount, StringComparison.OrdinalIgnoreCase));
+
             // Should not run bulk Update if any of the search parameters are invalid as it can lead to unpredicatable results
             await _searchService.ConditionalSearchAsync(request.ResourceType, searchParameters, cancellationToken, count: 1, logger: _logger);
             if (_contextAccessor.RequestContext?.BundleIssues?.Count > 0 && _contextAccessor.RequestContext.BundleIssues.Any(x => !string.Equals(x.Diagnostics, Core.Resources.TruncatedIncludeMessageForIncludes, StringComparison.OrdinalIgnoreCase)))
             {
                 throw new BadRequestException(_contextAccessor.RequestContext.BundleIssues.Select(issue => issue.Diagnostics).ToList());
             }
 
+            // Add back the maxCount to searchParameters
+            searchParameters.Add(Tuple.Create(KnownQueryParameterNames.MaxCount, request.MaxCount.ToString()));
```

Comment:
Why does this need to be added back if we are tracking it in the maximumNumberOfResourcesPerQuery property?